src/plugins/contextMenu/predefinedItems/columnRight.js

```diff
@@ -14,18 +15,29 @@ export default function columnRightItem() {
     },
     callback(key, normalizedSelection) {
       const latestSelection = normalizedSelection[Math.max(normalizedSelection.length - 1, 0)];
+      const selectedColumn = latestSelection?.end?.col;
+      // If there is no selection we have clicked on the corner and there is no data.
+      const columnsRight = isDefined(selectedColumn) ? selectedColumn + 1 : 0;
 
-      this.alter('insert_col', latestSelection.end.col + 1, 1, 'ContextMenu.columnRight');
+      this.alter('insert_col', columnsRight, 1, 'ContextMenu.columnRight');
     },
     disabled() {
       const selected = getValidSelection(this);
+      const anyCellVisible = this.countRows() > 0 && this.countCols() > 0;
 
-      if (!selected) {
+      if (!this.isColumnModificationAllowed()) {
         return true;
       }
-      if (!this.isColumnModificationAllowed()) {
+
+      // We have clicked on the corner and there is no data.
+      if (!anyCellVisible) {
+        return false;
+      }
```

Comment:
I don't think so. It's the case that there is no selection (so `getValidSelection` doesn't return it) and we've clicked on the corner. The `Insert column right` option should be visible. I will extend the comment. I hope that it will be helpful.

Reply:
The extended description within https://github.com/handsontable/handsontable/pull/6547/commits/cafe897983123c30ffaac2201176d8b2c3cd13ac#diff-9424ed687bf5720f6d531bb4c42eecfcL32-R33.

Reply:
IMO it's still not a "validSelection" so could be handled within the shared helper and do not repeat the code. Although, it's an implementation detail so I have to trust the engineer 👍